### PR TITLE
Add request/response body logging to HTTP tracer

### PR DESCRIPTION
**Pull request by DaveCTurner**
Adds another logger, `org.elasticsearch.http.HttpBodyTracer`, which logs the body of every HTTP request and response as well as the usual summaries.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
This would have been kind of useful in several support/debugging cases, and I hacked something together for my own use recently and then decided it wasn't far off production-ready anyway.

File: docs/reference/modules/network/tracers.asciidoc [hunk 1]
```diff
@@ -36,6 +36,42 @@ PUT _cluster/settings
 }
 --------------------------------------------------
 
+By default, the tracer logs a summary of each request and response which
```
Comment (tlrx) [inline docs/reference/modules/network/tracers.asciidoc:39]: I'm sure you already thought about it, but this is concerning if applied to REST APIs like Get API Key. We should maybe exclude sensitive APIs?

Comment (DaveCTurner) [reply docs/reference/modules/network/tracers.asciidoc:39]: It's a good question. I don't see that this feature differs much from the long-standing `ESLoggingHandler` which can be set to log the raw contents of every transport message. `TRACE` logs can be rather sensitive. But I would like other opinions on this.

Comment (DaveCTurner) [reply docs/reference/modules/network/tracers.asciidoc:39]: ^ particularly Tim's opinion (or at least Tim's opinion about whose opinion I should seek)

**Comment by DaveCTurner** [comment]
@tvernum any thoughts here?

File: docs/reference/modules/network/tracers.asciidoc (hunk 1 above)
Comment (tvernum) [reply docs/reference/modules/network/tracers.asciidoc:39]: We have some concept of filtering rest request (but not response) bodies today (*), however it's specific to security and only really used for audit logging.

I'm torn. It's only TRACE, it's off by default and even when it's on most of our log shipper configurations will drop TRACE messages, but it's one of those things that someone could raise as an issue later on. We probably need to get wider agreement first.

@jkakavas can we have a chat about this sometime?

(*) https://github.com/elastic/elasticsearch/blob/e059dd08f39b2197f4e81bb5368e869f71992dd6/server/src/main/java/org/elasticsearch/rest/RestRequestFilter.java#L33

**Review by tvernum — APPROVED** [review]
I'm happy with the protection the system property provides.

I haven't reviewed the mechanics of the logging though, so you'll want another review from @tlrx or @original-brownbear

**Review by tlrx — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
Thanks everyone! Great stuff.

**Pull request merged**